perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -5532,8 +5533,21 @@ public void deleteGroupsFromAutoRegistration(PerunSession sess, List<Group> grou
 	}
 
 	@Override
-	public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups) {
+	public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups) throws GroupNotAllowedToAutoRegistrationException {
 		for (Group group : groups) {
+			if (group.getName().equals(VosManager.MEMBERS_GROUP)) {
+				throw new GroupNotAllowedToAutoRegistrationException("Members group cannot be added to auto registration.", group);
+			}
+
+			try {
+				Attribute syncEnabledAttr = getPerunBl().getAttributesManagerBl().getAttribute(sess, group, GroupsManager.GROUPSYNCHROENABLED_ATTRNAME);
```

Comment:
There is also a group structure synchronization. It should be considered as well.
It will be probably a little more difficult because you will need to search also in the group tree. Check methods like `addMember`, specifically the use of method `isGroupInStructureSynchronizationTree`  in it.

Reply:
Ok, I've added groups with structure sync to the list of not allowed groups.